test/box-luatest/gh_8196_crash_box_internal_compact_test.lua

```diff
@@ -0,0 +1,12 @@
+local t = require('luatest')
+
+local g = t.group()
+
+local TARANTOOL_PATH = arg[-1]
+
+g.test_gh_8196 = function(_)
+    local code = "pcall(box.internal.compact, 1, 1); os.exit(0)"
+    local cmd = string.format('%s -e "%s"',
+                              TARANTOOL_PATH, code)
+    t.assert_equals(os.execute(cmd), 0)
```

Comment:
Could be called directly by this function, without `os.execute`. Also, the test should probably be located in the `app-luatest` suite.

Reply:
Re moving the test to `app-luatest`: come to think of it, it's a bad idea, since the test is about a box function. Still, I don't quite like `os.execute`. Let's start a test server instead and call the function from it, as we usually do?

Reply:
> > Re moving the test to app-luatest: come to think of it, it's a bad idea, since the test is about a box function.
> 
> I don't get why `box.internal.compact()` is a part of app-server and why directory for test is an important thing, but I moved it to `app-luatest`.

You misunderstood me: above I said that it's actually a bad idea to move the test to app-luatest (I changed my mind, sorry). Please move back.